dist/pythonlibs/riotctrl_shell/base.py

```diff
@@ -11,8 +11,69 @@
 """
 
 import abc
+import logging
 import pexpect
 import pexpect.replwrap
+import signal
+
+
+logger = logging.getLogger(__name__)
+
+
+class InterExpectREPLWrapper(pexpect.replwrap.REPLWrapper):
+    """
+    Same as pexpect.replwrap.REPLWrapper, only that expects to abort the
+    accumulation of further lines, `inter_expect`, can be provided to as a list
+    self.run_command()
+    """
+    def _expect_prompt(self, timeout=-1, async_=False, inter_expect=None):
+        expects = [self.prompt, self.continuation_prompt]
+        if inter_expect:
+            expects.extend(inter_expect)
+        return self.child.expect_exact(expects, timeout=timeout, async_=async_)
+
+    def run_command(self, command, timeout=-1, async_=False,
+                    inter_expect=None):
+        # Split up multiline commands and feed them in bit-by-bit
+        cmdlines = command.splitlines()
+        # splitlines ignores trailing newlines - add it back in manually
+        if command.endswith('\n'):
+            cmdlines.append('')
+        if not cmdlines:
+            raise ValueError("No command was given")
+
+        if async_:
+            from ._async import repl_run_command_async
+            return repl_run_command_async(self, cmdlines, timeout)
+
+        res = []
+        lpe = -1    # last prompt expect
+        self.child.sendline(cmdlines[0])
+        for line in cmdlines[1:]:
+            lpe = self._expect_prompt(timeout=timeout,
+                                      inter_expect=inter_expect)
+            res.append(self.child.before)
+            if lpe == 2:
+                break
+            self.child.sendline(line)
+
+        # Command was fully submitted, now wait for the next prompt
+        if lpe < 2:
+            lpe = self._expect_prompt(timeout=timeout,
+                                      inter_expect=inter_expect)
+            if lpe == 1:
+                # We got the continuation prompt - command was incomplete
+                self.child.kill(signal.SIGINT)
+                raise ValueError(
+                    "Continuation prompt found - input was incomplete:\n" +
+                    command
+                )
+        if lpe >= 2:
+            raise ValueError(
+                "Inter except found - input was incomplete:\n" +
+                command
+            )
```

Comment:
If a list is provided would be nice to know what was matched, e.g: an `assert`, `hardfault`, `etc`..

Reply:
Done